helix-term/src/commands.rs

```diff
@@ -1176,25 +1176,55 @@ fn goto_file_impl(cx: &mut Context, action: Action) {
     let primary = selections.primary();
     // Checks whether there is only one selection with a width of 1
     if selections.len() == 1 && primary.len() == 1 {
-        let count = cx.count();
-        let text_slice = text.slice(..);
-        // In this case it selects the WORD under the cursor
-        let current_word = textobject::textobject_word(
-            text_slice,
-            primary,
-            textobject::TextObject::Inside,
-            count,
-            true,
-        );
-        // Trims some surrounding chars so that the actual file is opened.
-        let surrounding_chars: &[_] = &['\'', '"', '(', ')'];
         paths.clear();
-        paths.push(
-            current_word
-                .fragment(text_slice)
-                .trim_matches(surrounding_chars)
-                .to_string(),
-        );
+
+        let is_valid_path_char = |c: &char| {
+            let valid_chars: &[char] = if cfg!(target_os = "windows") {
+                &[
+                    '@', '/', '\\', '.', '-', '_', '+', '#', '$', '%', '{', '}', '[', ']', ':',
+                    '!', '~', '=',
+                ]
+            } else {
+                &['@', '/', '.', '-', '_', '+', '#', '$', '%', '~', '=']
+            };
+            valid_chars.contains(c) || c.is_alphabetic()
+        };
+
+        let path = {
+            let start = {
+                let cursor_pos = primary.cursor(text.slice(..));
+
+                let pre_cursor_pos = cursor_pos.saturating_sub(1);
+                let post_cursor_pos = cursor_pos.saturating_add(1);
+
+                if is_valid_path_char(&text.char(cursor_pos)) {
+                    text.chars_at(cursor_pos)
+                } else if is_valid_path_char(&text.char(pre_cursor_pos)) {
+                    text.chars_at(pre_cursor_pos)
+                } else {
+                    text.chars_at(post_cursor_pos)
+                }
+            };
+
+            let head = start
+                .clone()
+                .reversed()
+                .take_while(is_valid_path_char)
+                .collect::<String>()
+                .chars()
+                .rev()
+                .collect::<String>();
+
+            let tail = start.take_while(is_valid_path_char).collect::<String>();
+
+            format!("{}{}", head, tail)
+        };
+        log::debug!("Goto file path: {}", path);
+
+        match shellexpand::full(&path) {
```

Comment:
I'm wondering if need to do shell expansion at all? We could pass the path through `expand_tilde` to take care of that common case. Full shell expansion seems like it might be overkill to me

Reply:
I didn't knew that there's `expand_tilde`. I removed the dependency and using `expand_tilde` now.